File: lib/layer2_core/light/e_surface_container_highest_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class ESurfaceContainerHighestLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl14;
      case 'standard': return cstl14;
      case 'blue': return cbl14;
      case 'green': return cgl14;
      case 'red': return crl14;
      case 'nothing': return cnol14;
      default: return cstl14;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/ESurfaceContainerHighestLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object ESurfaceContainerHighestLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl14
            AppTheme.STANDARD -> R.color.cstl14
            AppTheme.BLUE -> R.color.cbl14
            AppTheme.GREEN -> R.color.cgl14
            AppTheme.RED -> R.color.crl14
            AppTheme.NOTHING -> R.color.cnol14
        }
    }
}
